### ISSUE TITLE
Failed to remove portal

### ISSUE THREAD

**Issue by thorakmedichi** [gorhom/react-native-portal#11]
"@gorhom/portal": "^1.0.0"

I keep getting the following error in React Native

Failed to remove portal '1mrgf5_Wbp0kwl8wjzX1w', 'autocomplete_section_list' was not registered! 
    at PortalProviderComponent 
    
Here is the snippet of code where the issue occurs

```js
          <React.Fragment>
                <PortalHost name="autocomplete_section_list" />
                <View style={{ marginBottom: 10 }}>
                    <TextInput
                        autoFocus
                        isDisabled={isDisabled}
                        isRequired={isRequired}
                        label={label}
                        value={value}
                        placeholder={placeholder}
                        onChangeText={onChange}
                        onFocus={() => setHideResults(false)}
                        onBlur={onBlurTextInput}
                        startAdornment={startAdornment}
                        endAdornment={endAdornment}
                        style={style}
                    />
                    {
                        isFetching && (
                            <View style={[theme.selectList, styles.sectionList, styles.loading]}>
                                <Text style={theme.text}>Loading...</Text>
                            </View>
                        )
                    }
                    {
                        !hideResults && !isFetching && (
                            <Portal hostName="autocomplete_section_list">
                                <View style={styles.sectionList}>
                                    <SectionList
                                        sections={options}
                                        keyExtractor={(item) => item.label}
                                        renderItem={renderOption}
                                        renderSectionHeader={renderSectionHeader}
                                        keyboardShouldPersistTaps="always"
                                    />
                                </View>
                            </Portal>
                        )
                    }
                </View>
            </React.Fragment>
```

**Comment by mattlubner** [comment]
Any chance `hostName="autocomplete_section_list"` is being dynamically generated?

I was glancing at the code earlier and noticed Portal's useEffect for unmounting doesn't list any deps, meaning any changes to the Portal's `name` or `hostName` props could potentially be breaking the call to `removePortal`.

https://github.com/gorhom/react-native-portal/blob/master/src/components/portal/Portal.tsx#L39

**Comment by gorhom** [comment]
@thorakmedichi thanks for submitting this issue,

I have tried to reproduce the issue but i couldn't, would you mind add sample reproducible code from `/example` ? 

thanks

**Comment by gorhom** [comment]
@mattlubner yeah , i properly need to throw an error if a user updated any name ( for portal or host )

**Comment by thorakmedichi** [comment]
> @thorakmedichi thanks for submitting this issue,
> 
> I have tried to reproduce the issue but i couldn't, would you mind add sample reproducible code from `/example` ?
> 
> thanks

I'll try. It may be related to other portions of my code, ie this being in a dialog / modal etc. Its a bit hard / time consuming to get an exact replica of my issue without sharing my entire code base. But let me see what I can do...

Thanks for the quick replies

**Comment by galbenyosef** [comment]
I had this issue also,
this is how it resolved:

Ive modified conditional rendering wrapper Portal's Child to conditional rendering Portal itself,

was:
`<Portal hostName="info_tooltip">
                 {tooltip.index !== '' ? <InfoTooltip {...tooltip} /> : null}
              </Portal>`

has been modified to:
` {tooltip.index !== '' ? (
              <Portal hostName="info_tooltip">
                <InfoTooltip {...tooltip} />
              </Portal>
            ) : null}`

error is gone for me,

**Comment by zhigang1992** [comment]
We had this issue when Host and Portal are unmounted at the same time, I think in cases like this, instead of throwing an error, it should just be a no-op?

**Comment by gorhom** [comment]
this should be resolved in `v1.0.7`
